src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +30,365 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        dim = &sQuadInit[i].dim;
+        Matrix_MultVec3f(&dim->quad[0], &a);
+        Matrix_MultVec3f(&dim->quad[1], &b);
+        Matrix_MultVec3f(&dim->quad[2], &c);
+        Matrix_MultVec3f(&dim->quad[3], &d);
+        collider = &this->colliderQuad[i];
+        Collider_SetQuadVertices(collider, &a, &b, &c, &d);
+    }
+
+    Matrix_Pop();
+}
+
+void BgIkanaMirror_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgIkanaMirror* this = THIS;
+    Vec3f* vtx;
+    Vec3f vertices[3];
+    ColliderTrisElementInit* colliderElement;
+    s32 i;
+    s32 j;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerTempleMirrorCol);
+    Collider_InitTris(play, &this->colliderTris);
+    Collider_SetTris(play, &this->colliderTris, &this->dyna.actor, &sTrisInit, this->colliderTrisElements);
+    Matrix_SetTranslateRotateYXZ(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+
+    for (i = 0; i < ARRAY_COUNT(sTrisElementsInit); i++) {
+        colliderElement = &sTrisInit.elements[i];
+        vtx = &colliderElement->dim.vtx[0];
+        for (j = 0; j < 3; j++) {
+            Matrix_MultVec3f(vtx, &vertices[j]);
+            vtx++;
+        }
+        Collider_SetTrisVertices(&this->colliderTris, i, &vertices[0], &vertices[1], &vertices[2]);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        Collider_InitQuad(play, &this->colliderQuad[i]);
+        Collider_SetQuad(play, &this->colliderQuad[i], &this->dyna.actor, &sQuadInit[i]);
+    }
+
+    BgIkanaMirror_SetQuadVertices(this);
+    this->lightAbsorptionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightAbsorptionTexAnim);
+    this->lightEmissionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightEmissionTexAnim);
+    BgIkanaMirror_SetupWait(this);
+}
+
+void BgIkanaMirror_Destroy(Actor* thisx, PlayState* play) {
+    BgIkanaMirror* this = THIS;
+    s32 i;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(play, &this->colliderTris);
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        Collider_DestroyQuad(play, &this->colliderQuad[i]);
+    }
+}
+
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this) {
+    this->isEmittingLight = 0;
+    this->actionFunc = BgIkanaMirror_Wait;
+}
+
+void BgIkanaMirror_Wait(BgIkanaMirror* this, PlayState* play) {
+    s8 isEmittingLight;
+    s32 startEmittingLight = false;
+
+    // The light emission texture should gradually disappear from sight.
+    if (this->lightEmissionAlpha >= 101) {
```

Comment:
```suggestion
    if (this->lightEmissionAlpha > 100) {
```
as with the other one.

Reply:
Fixed in the next commit to be pushed.